src/ucp/wireup/wireup_cm.c

```diff
@@ -559,15 +584,34 @@ static unsigned ucp_cm_server_conn_request_progress(void *arg)
 }
 
 void ucp_cm_server_conn_request_cb(uct_listener_h listener, void *arg,
-                                   const char *local_dev_name,
-                                   uct_conn_request_h conn_request,
-                                   const uct_cm_remote_data_t *remote_data)
+                                   const uct_cm_listener_conn_req_cb_handle_t
+                                   *conn_req_handle)
 {
     ucp_listener_h ucp_listener = arg;
     uct_worker_cb_id_t prog_id  = UCS_CALLBACKQ_ID_NULL;
     ucp_conn_request_h ucp_conn_request;
+    uct_conn_request_h conn_request;
+    const uct_cm_remote_data_t *remote_data;
     ucs_status_t status;
 
+    if (!(conn_req_handle->field_mask & UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_CONN_REQUEST)) {
+        ucs_error("No connect request handle provided to the server (0x%lx)",
+                  conn_req_handle->field_mask);
+        return;
+    }
+
+    conn_request = conn_req_handle->conn_request;
+
+    if (!(ucs_test_all_flags(conn_req_handle->field_mask,
+                             (UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_REMOTE_DATA  |
+                              UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_LOCAL_DEV_NAME)))) {
```

Comment:
can we unite this with line 597?

Reply:
if one of the bits in lines 606-607 is missing, we can reject the request using the conn_request from above.